internal/pkg/agent/cmd/container_init_linux.go

```diff
@@ -0,0 +1,312 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+
+	"github.com/elastic/elastic-agent/internal/pkg/cli"
+	"github.com/elastic/elastic-agent/pkg/utils"
+)
+
+var (
+	// for unit-testing
+	capBound   = cap.GetBound
+	capGetFile = cap.GetFile
+	capProc    = cap.GetProc
+)
+
+// initContainer applies the following container initialisation steps:
+// - set any missing capabilities of Effective set based on the Bounding set at elastic-agent binary
+// - raise capabilities of the Ambient set to match the Effective set
+// - chown all agent-related paths if DAC_OVERRIDE capability is not in the Effective set
+// If new binary capabilities are set then the returned cmd will be not nil. Note that it is up to caller to invoke
+// the returned cmd and spawn an agent instance with all the capabilities.
+func initContainer(streams *cli.IOStreams, skipFileCapabilities bool) (shouldExit bool, err error) {
+	isRoot, err := utils.HasRoot()
+	if err != nil {
+		return true, err
+	}
+	if !skipFileCapabilities && !isRoot {
+		executable, err := os.Executable()
+		if err != nil {
+			return true, err
+		}
+
+		logInfo(streams, "agent container initialisation - file capabilities")
+		updated, err := updateFileCapsFromBoundingSet(executable)
```

Comment:
Could this be done with out the `--skip-file-capabilities` flag? Like determine which file caps and missing and apply them and then re-exec? Now the new re-exec will do the same thing and see that it has all the capabilities and continue on?

Would that be bettter and cleaner?

Reply:
I totally agree with skipping this flag completely, it was just my just-in-case thinking that made me add this flag if somebody wanted to skip this step for any reason 🥲 but I will remove it

Reply:
Can you think of a reason they would want to skip it?

They can still choose to run as root and it will still work the same, correct?

Reply:
>Can you think of a reason they would want to skip it?

Tbh I don't have any reason reason in mind 

>They can still choose to run as root and it will still work the same, correct?

Yep 100% correct, running agent with root equals that we won't deal with agent capabilities at all

Reply:
I ve removed the flag, and I have manually tested that this doesn't affect the flow, feel free to unresolve if you spot anything